File: src/main/java/co/utb/softeng/moviesapp/services/ActorService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.utb.softeng.moviesapp.services;

import co.utb.softeng.moviesapp.entities.Actor;
import java.util.List;

/**
 *
 * @author William
 */
public interface ActorService {
    
    public List<Actor> getAllActors();
    public Actor getActorById(Long id);
    public List<Actor> getActorByName(String name);
    public List<Actor> getActorByMovieName(String movieName);
    public void saveOrUpdateActor(Actor actor);
    public void deleteActor(Long id);
    
    
}

File: src/main/webapp/resources/assets/js/app/services/actorService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

var servicesModule = angular.module('springMoviesAppServices');
servicesModule.factory('actorService', ['$http', function ($http) {
        return {
            apiUrl: apiUrl,
            getAllActors: function () {
                return $http.get(this.apiUrl + 'actor/');
            },
            getActorById: function (actorId) {
                return $http.get(this.apiUrl + 'actor/' + actorId);
            },
            createActor: function (actor) {
                return $http.put(this.apiUrl + 'actor/', actor);
            }
        };
    }]);


